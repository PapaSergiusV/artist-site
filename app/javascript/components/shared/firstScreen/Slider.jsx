import React, { useState, useRef, useEffect } from "react";
import CSSModules from "react-css-modules";

import { showWindow } from "../../../libs/animation";
import styles from "./Slider.module.scss";

const Slider = props => {
  const wrapper = useRef(null);
  const { slides, lifeTime = 10000 } = props;
  const [id, setSlide] = useState(0);

  useEffect(() => {
    if (slides && slides.length > 1) {
      let nextId = id + 1;
      if (nextId >= slides.length)
        nextId = 0;
      let nextSlide = document.createElement("div");
      nextSlide.style.background = `url(${slides[nextId]})`;
      nextSlide.style.height = "100%";
      nextSlide.style.width = "100%";
      nextSlide.style.backgroundPosition = "center";
      nextSlide.style.position = "absolute";
      nextSlide.style.top = "0";
      setTimeout(() => {
        wrapper.current.appendChild(nextSlide);
        showWindow(nextSlide);
      }, lifeTime - 150);
      setTimeout(() => {
        nextSlide.remove();
        setSlide(nextId);
      }, lifeTime);
    }
  });

  if (!slides || slides.length <= 1)
    return (
      <div styleName="wrapper" ref={wrapper}>
        <div styleName="slide"></div>
      </div>
    );
  return (
    <div styleName="wrapper" ref={wrapper}>
      <div styleName="slide" style={{backgroundImage: `url(${slides[id]})`}}></div>
    </div>
  );
};

export default CSSModules(Slider, styles);