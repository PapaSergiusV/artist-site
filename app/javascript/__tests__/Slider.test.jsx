import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import "../setupTests";

import Slider from "../components/shared/firstScreen/Slider";

describe("MainMenu component", () => {
  it("should render slider without props", () => {
    const wrapper = shallow(<Slider />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});