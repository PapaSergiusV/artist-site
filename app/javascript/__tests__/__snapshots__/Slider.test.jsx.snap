// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu component should render slider without props 1`] = `
<div
  className="wrapper"
>
  <div
    className="slide"
  />
</div>
`;
